src/Container.php

```diff
@@ -228,6 +228,10 @@ public function delegate(InteropContainerInterface $container)
     {
         $this->delegates[] = $container;
 
+        if ($container instanceof ContainerAwareInterface || $container instanceof ImmutableContainerAwareInterface) {
```

Comment:
I think this only need to be `ImmutableContainerAwareInterface` as that's the one that covers container-interop and effectively we're only supporting that for delegates, thoughts? @localheinz